### PR TITLE
Internal: Adjust kit library filter by subscription plan [ED-15792]

### PR DESCRIPTION
**Pull request by ronenboxer**

### CODE REVIEW

File: app/modules/kit-library/assets/js/models/taxonomy.js [hunk 1]
```diff
@@ -1,6 +1,45 @@
...
+	[ AGENCY ]: __( 'Agency', 'elementor' ),
+};
+
+export const NEW_PLAN_TEXTS = {
+	[ FREE ]: __( 'Free', 'elementor' ),
+	[ ESSENTIAL ]: __( 'Essential', 'elementor' ),
+	[ ADVANCED ]: __( 'Advanced & Higher', 'elementor' ),
+};
+
+const TAXONOMY_TRANSFORM_MAP = {
```
Comment (mykytamurzin) [inline app/modules/kit-library/assets/js/models/taxonomy.js:28]: is this a different TAXONOMIES?  
```

export const TAXONOMIES = [
	{
		key: 'categories',
```

File: app/modules/kit-library/assets/js/hooks/use-kits.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const KEY = 'kits';
 export const defaultQueryParams = {
 	favorite: false,
 	search: '',
-	taxonomies: taxonomyType.reduce( ( current, { key } ) => {
+	taxonomies: TAXONOMIES.reduce( ( current, { key } ) => {
```
Comment (mykytamurzin) [inline app/modules/kit-library/assets/js/hooks/use-kits.js:17]: strange naming here

File: app/modules/kit-library/assets/js/models/taxonomy.js (hunk 1 above)
Comment (ronenelementor) [reply app/modules/kit-library/assets/js/models/taxonomy.js:28]: WDYM?

File: app/modules/kit-library/assets/js/hooks/use-kits.js (hunk 2 above)
Comment (ronenelementor) [reply app/modules/kit-library/assets/js/hooks/use-kits.js:17]: Should I just lowercase it? Something else?

File: app/modules/kit-library/assets/js/models/taxonomy.js (hunk 1 above)
Comment (ronenelementor) [reply app/modules/kit-library/assets/js/models/taxonomy.js:28]: We needed to take the original 5 plans, and divide it into 3 new ones

File: app/modules/kit-library/assets/js/components/taxonomies-filter.js [hunk 3]
```diff
@@ -12,12 +12,35 @@ export default function TaxonomiesFilter( props ) {
...
 
-			return taxonomyType
-				.map( ( tagType ) => ( {
-					...tagType,
-					data: props.taxonomies.filter( ( item ) => item.type === tagType.key ),
-				} ) )
-				.filter( ( { data } ) => data.length > 0 );
+			const taxonomies = [ ...props.taxonomies ];
+
+			return TAXONOMIES.reduce( ( carry, taxonomyItem ) => {
```
Comment (mykytamurzin) [inline app/modules/kit-library/assets/js/components/taxonomies-filter.js:17]: also something like 
```
return TaxonomyNames.getFormatted( ... ) 
```

**Review by murzin-nikita — APPROVED** [review]

**Review by baghdasarovelementor — APPROVED** [review]

**Pull request merged**